0003-styling-with-functions/Styling with Functions.playground/Pages/New.xcplaygroundpage/Contents.swift

```diff
@@ -209,18 +209,19 @@ final class SignInViewController: UIViewController {
 
     self.view.addSubview(gradientView)
     self.view.addSubview(rootStackView)
-
-    NSLayoutConstraint.activate([
-      gradientView.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor),
-      gradientView.leadingAnchor.constraint(equalTo: self.view.leadingAnchor),
-      gradientView.trailingAnchor.constraint(equalTo: self.view.trailingAnchor),
-      gradientView.bottomAnchor.constraint(equalTo: self.view.centerYAnchor),
-
-      rootStackView.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor),
-      rootStackView.leadingAnchor.constraint(equalTo: self.view.leadingAnchor),
-      rootStackView.trailingAnchor.constraint(equalTo: self.view.trailingAnchor),
-      ])
-  }
+    if #available(iOS 11, *) {
+        NSLayoutConstraint.activate([
+            gradientView.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor),
```

Comment:
Instead of the `#available` check, can we use `self.topLayoutGuide` instead? I think this uses the `safeAreaLayoutGuide` automatically.

Reply:
`topLayoutGuide` is [deprecated since iOS 11](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621367-toplayoutguide). I think a combination of #available and `topLayoutGuide` would make sense?

    if #available(iOS 11, *) {
        NSLayoutConstraint.activate([
            gradientView.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor),
            gradientView.leadingAnchor.constraint(equalTo: self.view.leadingAnchor),
            gradientView.trailingAnchor.constraint(equalTo: self.view.trailingAnchor),
            gradientView.bottomAnchor.constraint(equalTo: self.view.centerYAnchor),
            
            rootStackView.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor),
            rootStackView.leadingAnchor.constraint(equalTo: self.view.leadingAnchor),
            rootStackView.trailingAnchor.constraint(equalTo: self.view.trailingAnchor),
            ])
    } else {
        NSLayoutConstraint.activate([
            gradientView.topAnchor.constraint(equalTo: self.topLayoutGuide.topAnchor),
            gradientView.leadingAnchor.constraint(equalTo: self.view.leadingAnchor),
            gradientView.trailingAnchor.constraint(equalTo: self.view.trailingAnchor),
            gradientView.bottomAnchor.constraint(equalTo: self.view.centerYAnchor),
            
            rootStackView.topAnchor.constraint(equalTo: self.topLayoutGuide.topAnchor),
            rootStackView.leadingAnchor.constraint(equalTo: self.view.leadingAnchor),
            rootStackView.trailingAnchor.constraint(equalTo: self.view.trailingAnchor),
            ])
    }

either way, I submitted the PR to make the playground work on Xcode 10 beta 3, but now on beta 4 the whole thing crashes on my end. This might be a mute change for now anyways.

Reply:
Seems like the main issue is incremental compilation is a lot pickier with order of definition these days. I just pushed [this commit](https://github.com/pointfreeco/episode-code-samples/commit/e2f3dcb932bf341d5be5b68f0a646d9bc942dd9c) so that `RoundedButton` is defined before `FilledButton` and was able to compile on Xcode 10 without any additional changes. Can you try pulling down the latest to see if you have the same experience?

Reply:
Yep, it works. 👍